src/test_application/activescenes/scene_vehicles.cpp

```diff
@@ -477,22 +494,18 @@ Session setup_vehicle_spawn_vb(
 
                 std::advance(itRemapMachOffset, 1);
             }
-
         }
+#endif
 
+        // Copy Machines and Nodes from VehicleBuilder to Scene
 
         auto itRemapPartOffset = std::begin(rVehicleSpawnVB.m_remapPartOffsets);
 
         for (VehicleData const* pVData : rVehicleSpawnVB.m_dataVB)
         {
-            if (pVData == nullptr)
-            {
-                continue;
-            }
-
-            auto remapPart = arrayView(rVehicleSpawnVB.m_remapPartAll).exceptPrefix(*itRemapPartOffset);
+            auto remapPart = arrayView(rVehicleSpawnVB.m_remapParts).exceptPrefix(*itRemapPartOffset);
+            std::advance(itRemapPartOffset, 1);
 
-            // remapMach[source MachAnyId] -> destination MachAnyId
             std::vector<MachAnyId> remapMach(pVData->m_machines.m_ids.capacity());
```

Comment:
Doesn't this intantiate a bunch of values with "default" construction?

You sure that's the right constructor?

Reply:
This is right; I need a vector with a given size. The default values aren't used though, they technically just be garbage.